File: code/perfect-java-code/perfect-design-pattern/src/creation/factory/simplefactory/pizzastore/pizza/GreekPizza.java
package creation.factory.simplefactory.pizzastore.pizza;

/**
 * @author Kelly
 * @create 2020-09-18 16:35
 */
public class GreekPizza extends Pizza {
    @Override
    public void prepare() {
        System.out.println("给【希腊披萨】准备原材料...");
    }
}

File: code/perfect-java-code/perfect-scala/src/main/scala/com/learn/chapter17/simplefactory/pizzastore/pizza/GreekPizza.scala
package com.learn.chapter17.simplefactory.pizzastore.pizza

/**
 * @author Kelly
 * @create 2021-05-13 15:37
 */
class GreekPizza extends Pizza {
  override def prepare(): Unit = {
    this.name = "Greek Pizza"
    println(this.name + " preparing...")
  }
}
